### DISCUSSION TITLE
Not able to start using Avalonia in VSCode on Linux

### DISCUSSION THREAD

**Discussion by nsk7even** [AvaloniaUI/Avalonia#15814] [Q&A]
Hey there, I am a professional C# developer, using Microsoft Visual Studio on Windows at business.
Privately I only use Linux. Today I started to try out Avalonia for VSCode to build a Linux UI application.

Sadly, my experience with Avalonia is quite bad until now:

1. Installed `net8`
2. Installed `Avalonia for VSCode`
3. `dotnet new install Avalonia.Templates`
4. _Create Avalonia Project_
5. _Avalonia MVVM App_
6. _Open_
7. _Start Debugging_
8. Select `C#` in popup
9. Select `C# <projectname>` in popup
10. Hello world example running
11. Terminal error `CS0103` of `Views/MainWindow.axaml.cs(9,9)`: _the name "InitializeComponent" does not exists_ (compiles anyways)
12. Console is flooded with _[..] Skipped loading symbols.[..]_ messages
13. Inserted `"moduleLoad": false` for `logging` in `launch.json` to stop this
14. Console shows _ObjectDisposedExceptions_ at stop

So far so good (except the ugly terminal outputs). But it seems that this example is quite of precompiled/prepared, because as soon as I am making changes, the hell begins:

1. Text completion does not work (see https://github.com/AvaloniaUI/AvaloniaVSCode/issues/113 and https://github.com/AvaloniaUI/AvaloniaVSCode/issues/110)
2. Downgraded Avalonia for VSCode to `0.29`
3. Text completion works now
4. Made change to `MainWindow.axaml`: wrapped TextBlock from example into a Grid
5. Build failure: `CSC : error AXN0002: Dots aren't allowed in type names Line 20, position 14. [.... /dev/test/test.csproj]`

Well. Starting with troubleshooting after troubleshooting when just trying to set up the toolchain is not what my intent was.

Am I doing something completely wrong?
Is it too early for beginners to start using Avalonia UI on linux?

----------------------------------

Sorry, just recognised that the build failure was due to a syntax error. This wasn't visualized in the UI...

**Comment by timunie** [comment]
Rider can maybe run on Linux. Dev experience is way better there imo when it comes to Avalonia

**Reply by nsk7even** [reply]
Yes, Rider provides a far better experience, I tried it out in parallel.
But it is much slower and costs money. On a faster machine and when using it regularly, it is the better option for sure.
